org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/snippets/XMLDeclarationSnippetContext.java

```diff
@@ -37,8 +37,8 @@ public boolean isMatch(ICompletionRequest request, Map<String, String> model) {
 			// --> <?xml version="1.0" encoding="UTF-8" | ?>
 			return false;
 		}
-		// check if document already defined a xml processing instruction.
-		if (hasProlog(document, offset)) {
+		// check if document already defined a xml declaration.
+		if (hasProlog(document, offset) || (document.hasProlog() && offset == 0)) {

```

Comment:
done.

code looks a bit different than your suggestion, though, because there's no `getProlog()` (which could be added, but I wasn't sure if I should do that) and the refactored `inProlog(node, offset)` takes the document, not the node (which again, could be changed, but I didn't want to touch on that).